internal/analytics/client.go

```diff
@@ -155,17 +157,22 @@ func Capture(eventType string, providers registry.Providers, data Message, diags
 	if c.disabled {
 		return nil
 	}
+
 	eventProps := map[string]interface{}{
 		"version":     c.version.Version,
 		"commitId":    c.version.CommitId,
 		"buildDate":   c.version.BuildDate,
 		"env":         c.env,
-		"data":        data.Properties(),
 		"instanceId":  c.instanceId,
 		"success":     !diags.HasErrors(),
 		"providers":   providers,
 		"diagnostics": core.SummarizeDiagnostics(diags),
 	}
+
+	if data != nil && !reflect2.IsNil(data) {
```

Comment:
```suggestion
	if !reflect2.IsNil(data) {
```

`reflect2.IsNil` already does the `data != nil` check